include/pmacc/assert.hpp

```diff
@@ -24,35 +24,51 @@
 
 #include "pmacc/debug/abortWithError.hpp"
 
+#include <cassert>
+
 #ifdef NDEBUG
 // debug mode is disabled
 
 /* `(void)0` force a semicolon after the macro function */
 #    define PMACC_ASSERT(expr) ((void) 0)
 
+/* `(void)0` force a semicolon after the macro function */
+#    define PMACC_DEVICE_ASSERT(expr) ((void) 0)
+
 /* `(void)0` force a semicolon after the macro function */
 #    define PMACC_ASSERT_MSG(expr, msg) ((void) 0)
 
 #else
 
 // debug mode is enabled
 
-/** assert check
+/** assert check (host side only)
  *
- * if `NDEBUG` is not defined: macro expands to (void)0
+ * if `NDEBUG` is defined: macro expands to (void)0
  *
  * @param expr expression to be evaluated
  */
 #    define PMACC_ASSERT(expr) (!!(expr)) ? ((void) 0) : pmacc::abortWithError(#    expr, __FILE__, __LINE__)
 
-/** assert check with message
+/** assert check with message (host side only)
  *
- * if `NDEBUG` is not defined: macro expands to (void)0
+ * if `NDEBUG` is defined: macro expands to (void)0
  *
  * @param expr expression to be evaluated
  * @param msg output message (of type `std::string`) which is printed if the
  *            expression is evaluated to false
  */
 #    define PMACC_ASSERT_MSG(expr, msg) (!!(expr)) ? ((void) 0) : pmacc::abortWithError(#    expr, __FILE__, __LINE__, msg)
 
+/** assert check for kernels (device side)
+ *
+ * if `NDEBUG` is defined: macro expands to (void)0
+ * in kernel asserts are supported only for cuda device compatibility >= 2.
+ * @param expr expression to be evaluated
+ */
+#    if((BOOST_LANG_CUDA && PMACC_CUDA_ARCH < 200))
+#        define PMACC_DEVICE_ASSERT(expr) ((void) 0)
```

Comment:
You can remove rhe version for architectures smaller than 2.0  PIConGPU is not supporting anything older rhan architecture 2.0.
Please also adjust the comment